src/Bicep.Core.UnitTests/TypeSystem/MicrosoftGraph/MicrosoftGraphResourceTypeProviderTests.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Extensions;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.Workspaces;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using FluentAssertions.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bicep.Core.UnitTests.TypeSystem.MicrosoftGraph
+{
+    [TestClass]
+    public class MicrosoftGraphResourceTypeProviderTests
+    {
+        private static string MicrosoftGraphBuildInName = "microsoftGraph";
+
+        private static NamespaceType GetMicrosoftGraphNamespaceType()
+        {
+            return BicepTestConstants.NamespaceProvider.TryGetNamespace(MicrosoftGraphBuildInName, MicrosoftGraphBuildInName, ResourceScope.ResourceGroup, BicepTestConstants.Features, BicepSourceFileKind.BicepFile, null)!;
+        }
+
+        [TestMethod]
+        public void MicrosoftGraphResourceTypeProvider_can_list_all_types_without_throwing()
+        {
+            var resourceTypeProvider = GetMicrosoftGraphNamespaceType().ResourceTypeProvider;
+            var availableTypes = resourceTypeProvider.GetAvailableTypes();
+
+            // sanity check - we know there should be a lot of types available
+            var maxExpectedTypes = 5;
```

Comment:
Should we do the opposite to check that there are at least 5 types? In the future, if we add more types, the test will fail.

Reply:
Yeah that makes more sense. I will update the test